CMakeLists.txt

```diff
@@ -80,11 +84,15 @@ if (UNIX)
   add_compile_options(
     -Wno-uninitialized
     $<$<COMPILE_LANGUAGE:Fortran>:-std=legacy>
+    $<$<COMPILE_LANGUAGE:Fortran>:-cpp>
     $<$<COMPILE_LANGUAGE:Fortran>:-fno-second-underscore>
   )
   if (APPLE)
+    set(is_fortran "$<COMPILE_LANGUAGE:Fortran>")
+    set(is_greater_v10 "$<VERSION_GREATER:$<Fortran_COMPILER_VERSION>,10.0.0>")
+    set(meet_requirements "$<AND:${is_fortran},${is_greater_v10}>")
```

Comment:
instead of `meet_requirements` this should be called `is_gfortran_greater_v10` to make the code more readable

Reply:
In general, I think it is not good style to introduce a named variable for something that is used only once and in one place. Assuming that whitespace is ignored, can you perhaps write this requirement over several lines to make it more readable without introducing extra variables?

Code which may or may not work:
```
add_compile_options(
  $<
    $<AND:
        $<COMPILE_LANGUAGE:Fortran>,
        $<VERSION_GREATER:$<Fortran_COMPILER_VERSION>,10.0.0>
    >:-fallow-argument-mismatch
   >
)
```
If splitting the requirement over several lines is not possible then I agree that introducing variables is better.

Reply:
Agree.